sonar-ldap-plugin/src/main/java/org/sonar/plugins/ldap/LdapContextFactory.java

```diff
@@ -92,7 +97,25 @@ public InitialDirContext createUserContext(String principal, String credentials)
   }
 
   private InitialDirContext createInitialDirContext(String principal, String credentials, boolean pooling) throws NamingException {
-    return new InitialLdapContext(getEnvironment(principal, credentials, pooling), null);
+    final InitialLdapContext ctx;
+    if (startTLS) {
+      ctx = new InitialLdapContext(getEnvironment(null, null, /* TODO(Godin): connection pooling requires proper TLS shutdown */false), null);
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Complete the task associated to this TODO comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1135)